app/views/tag/_form.html.erb

```diff
@@ -3,12 +3,18 @@
 <form id="tagform" class="form" data-remote="true" action="<%= url %>">
   <div class="control-group">
     <input class="form-control" name="remote" type="hidden" value="true" />
-    <div class="input-group">
-      <input aria-label="Enter tag name" autocomplete="off" class="tag-input form-control" name="name" type="text" placeholder="<%= translation('tag._tagging.enter_tags',{},false) %>" data-provide="typeahead" />
-      <div class="input-group-append">
-        <%= render partial: 'tag/advanced_tagging' %>
+    <% if @node && @node.has_tag('locked') && (current_user.role != "admin" && current_user.role != "moderator") %>
```

Comment:
Hi Manasa, in this line `(current_user.role != "admin" && current_user.role != "moderator")`, a current_user may not be part of both roles, so if a user is an admin and not a moderator, the logic will be false but I think you need this code to only display if a user is either not an admin or moderator...

please have a look at the code below that does a great job with this...
<img width="758" alt="Screenshot 2021-06-09 at 11 55 25" src="https://user-images.githubusercontent.com/7622875/121325067-f82d0f00-c919-11eb-8efe-3e00024a9944.png">

It is part of https://github.com/publiclab/plots2/issues/9739


so you want to check that the current_role does not include either of those roles

Reply:
Hi @RuthNjeri. `(current_user.role != "admin" && current_user.role != "moderator")` is false when the user is an admin and not a moderator, which is why the message `Only admins / moderators can add tags on locked content.` will not be displayed. That message will be displayed when the user is **neither an admin nor a moderator** because both admins and moderators are allowed to add tags on a 'locked' note.

Reply:
oh, I see, thanks for the clarification @Manasa2850